fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(data => this.setState({ todos: data }))
+      .catch(err => console.error(err));
+  }
+
+  addTodo(newTodo) {
+    const cloneList = [...this.state.todos];
+    newTodo.todoId = (cloneList[cloneList.length - 1].todoId + 1);
+    cloneList.push(newTodo);
+    fetch('/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo)
+    })
+      .then(res => res.json())
+      .then(() => this.setState({ todos: cloneList }));
```

Comment:
Here you want to capture the returned data from the server in your parameter list, it should be the new todo item from the database with the generated id. Until you get a response from the server it hasn't happened yet, the server and more directly the database is the single source of truth. You never want to update your front end code until you have confirmation from the server everything worked out. With the new todo item from the server you can now push to your `cloneList` and then update state.